crates/sdl-encoder/src/lib.rs

```diff
@@ -0,0 +1,373 @@
+        }
+
+        write!(f, "schema {{")?;
+        write!(f, "{}", fields)?;
+        writeln!(f, "\n}}")
+    }
+}
+
+/// Define whether a Field is a vector.
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub enum VecState {
+    /// The type is not a vector: e.g. `String`.
+    None,
+    /// The type is a nullable vector: e.g. `[String]`.
+    Nullable,
+    /// The type is a non nullable vector: e.g. `[String]!`.
+    NonNullable,
+}
+/// Field in a given SDL type.
+#[derive(Debug, PartialEq, Clone)]
+pub struct Field {
+    description: Option<String>,
+    name: String,
+    type_: String,
+    default: Option<String>,
+    is_nullable: bool,
+    vec_state: VecState,
+}
+
+impl Field {
+    /// Create a new instance of Field.
+    pub fn new(name: String, type_: String) -> Self {
+        Self {
+            description: None,
+            name,
+            type_,
+            default: None,
+            is_nullable: true,
+            vec_state: VecState::None,
+        }
+    }
+
+    /// Set the field's description.
+    pub fn description(&mut self, description: String) {
+        self.description = Some(description);
+    }
+
+    /// Set the field's default.
+    pub fn default(&mut self, default: String) {
+        self.default = Some(default);
+    }
+
+    /// Set the field's is nullable.
+    pub fn is_nullable(&mut self, nullable: bool) {
+        self.is_nullable = nullable;
+    }
+
+    /// Set the field's vec state.
+    pub fn vec_state(&mut self, vec_state: VecState) {
+        self.vec_state = vec_state;
+    }
+}
+
+impl Display for Field {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if let Some(description) = &self.description {
+            // Let's indent description on a field level for now, as all fields
+            // are always on the same level and are indented by 2 spaces.
+            writeln!(f, "  \"\"\"\n  {}\n  \"\"\"", description)?;
+        }
+        let null = if self.is_nullable { "!" } else { "" };
+        let type_ = match self.vec_state {
+            VecState::None => format!("{}{}", &self.type_, null),
+            VecState::Nullable => format!("[{}{}]", &self.type_, null),
+            VecState::NonNullable => format!("[{}{}]!", &self.type_, null),
+        };
+        let default = if let Some(default) = &self.default {
+            match self.vec_state {
+                VecState::None => format!(" = {}{}", default, null),
+                VecState::Nullable => format!(" = [{}{}]", default, null),
+                VecState::NonNullable => format!(" = [{}{}]!", default, null),
+            }
+        } else {
+            String::new()
+        };
+        // TODO(@lrlna): double check with folks if it's a valid SDL if the last
+        // field in a type has a comma. If not, we can move the 'comma logic' to
+        // TypeDef/SchemaDef Display implementations.
+        write!(f, "  {}: {}{},", &self.name, type_, default)
```

Comment:
Can the last field in a type definition have a comma, like so:
```js
type Query {
  key: String,
  key2: Int,
  key3: String, // <--- this comma here valid?
}
```

Reply:
@lrlna It can have a comma there, yes, but also _all the commas are optional_.

Reply:
eh?????? 

I'll drop the comma then, easier for me \o/

Reply:
It's pretty weird, but commas are treated as whitespace in the GraphQL grammar, so you can insert them anywhere you want. Even `type,Query,{key:,String,},}` would be valid…